laplace/baselaplace.py

```diff
@@ -1568,8 +1567,1025 @@ def sample(self, n_samples=100, generator=None):
+            return -torch.einsum('bcp,p->bc', Js, self.prior_mean - self.map_estimate)
+
+    def log_marginal_likelihood(self, prior_precision=None, sigma_noise=None):
+        """Compute the Laplace approximation to the log marginal likelihood.
+        Requires that the Laplace approximation has been fit before.
+        The resulting torch.Tensor is differentiable in `prior_precision` and
+        `sigma_noise` if these have gradients enabled.
+        By passing `prior_precision` or `sigma_noise`, the current value is
+        overwritten. This is useful for iterating on the log marginal likelihood.
+
+        Parameters
+        ----------
+        prior_precision : torch.Tensor, optional
+            prior precision if should be changed from current `prior_precision` value
+        sigma_noise : [type], optional
+            observation noise standard deviation if should be changed
+
+        Returns
+        -------
+        log_marglik : torch.Tensor
+        """
+        # update prior precision (useful when iterating on marglik)
+        if prior_precision is not None:
+            self.prior_precision = prior_precision
+
+        # update sigma_noise (useful when iterating on marglik)
+        if sigma_noise is not None:
+            if self.likelihood != 'regression':
+                raise ValueError('Can only change sigma_noise for regression.')
+            self.sigma_noise = sigma_noise
+
+        return self.log_likelihood - 0.5 * (self.log_det_ratio + self.scatter)
+
+    @property
+    def prior_precision(self):
+        return self._prior_precision
+
+    @prior_precision.setter
+    def prior_precision(self, prior_precision):
+        self._posterior_scale = None
+        if np.isscalar(prior_precision) and np.isreal(prior_precision):
+            self._prior_precision = torch.tensor([prior_precision], device=self._device)
+        elif torch.is_tensor(prior_precision):
+            if prior_precision.ndim == 0:
+                # make dimensional
+                self._prior_precision = prior_precision.reshape(-1).to(self._device)
+            elif prior_precision.ndim == 1:
+                if len(prior_precision) not in [1, self.n_layers, self.n_params]:
+                    raise ValueError(
+                        'Length of prior precision does not align with architecture.'
+                    )
+                self._prior_precision = prior_precision.to(self._device)
+            else:
+                raise ValueError(
+                    'Prior precision needs to be at most one-dimensional tensor.'
+                )
+        else:
+            raise ValueError(
+                'Prior precision either scalar or torch.Tensor up to 1-dim.'
+            )
+        # This is a change from BaseLaplace. If the prior precision is changed, the cholesky
+        #  decomposition needs to be recomputed.
+        self._recompute_Sigma = True
+
+    def state_dict(self) -> dict:
```

Comment:
Please add a test case for saving & loading `FunctionalLaplace` in `tests/test_serialization.py`